langroid/agent/chat_agent.py

```diff
@@ -623,6 +799,139 @@ def disable_message_use_except(self, message_class: Type[ToolMessage]) -> None:
             self.llm_tools_usable.discard(r)
             self.llm_functions_usable.discard(r)
 
+    def _load_output_format(self, message: ChatDocument) -> None:
+        """
+        If set, attempts to parse a value of type `self.output_format` from the message
+        contents and assigns it to `content_any`.
+        """
+        if self.output_format is not None:
```

Comment:
This `_load_output_format` ends up getting called twice - once in `llm_response_messages` and then at the end of `llm_response` - this works fine, but it does some (minor) un-necessary work by trying to populate `content_any` all over again. I wonder if we should check if it already exists, and skip, or maybe just keep it as is just to be safe?